pkg/controller/utils/machine_helper_test.go

```diff
@@ -0,0 +1,280 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/openshift/cloud-ingress-operator/pkg/testutils"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	awsproviderapi "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1beta1"
+)
+
+func TestUpdateAWSLBList(t *testing.T) {
+	clustername := "lbupdatetest"
+	sampleMachine := testutils.CreateMachineObj("master-0", clustername, "master", testutils.DefaultRegionName, testutils.DefaultAzName)
+	objs := []runtime.Object{&sampleMachine}
+	mocks := testutils.NewTestMock(t, objs)
+
+	oldLBList := []awsproviderapi.LoadBalancerReference{
+		{
+			// <clustername>-<id>-ext
+			Name: fmt.Sprintf("%s-%s-ext", clustername, "12345"),
+			Type: awsproviderapi.NetworkLoadBalancerType,
+		},
+		{
+			// <clustername>-<id>-int
+			Name: fmt.Sprintf("%s-%s-int", clustername, "12345"),
+			Type: awsproviderapi.NetworkLoadBalancerType,
+		},
+	}
+	newLBList := []awsproviderapi.LoadBalancerReference{
+		{
+			// Just something random!
+			Name: fmt.Sprintf("%s-%s-test", clustername, "12345"),
+			Type: awsproviderapi.NetworkLoadBalancerType,
+		},
+	}
+	// quickly make sure the test is going to measure an actual change
+	if len(oldLBList) != 2 || len(newLBList) != 1 {
```

Comment:
not sure this is needed?

Reply:
The argument against is "Well, don't change the lists" and I hear that. The purpose of the assertion here is to ensure that the tested code is capable of going from two to one. But say the `oldLBList` is of length 1, or even 0, the test is going to behave differently.

I guess that's a roundabout way of saying the test should be more flexible with multiple test cases to _validate_ the different behaviour, eh?